commit b32aa6fb2e788d237799fd52e095b098f5c2eee0
Author: Daniel Muey <dan@cpanel.net>
Date:   Mon Mar 26 08:32:22 2018 -0500

    Apply 0001-Increase-random-seed-size.patch

diff --git a/server/core.c b/server/core.c
index 5b917dd..a05f67d 100644
--- a/server/core.c
+++ b/server/core.c
@@ -5331,7 +5331,7 @@ AP_CORE_DECLARE(void) ap_random_parent_after_fork(void)
 
 AP_CORE_DECLARE(void) ap_init_rng(apr_pool_t *p)
 {
-    unsigned char seed[8];
+    unsigned char seed[1024];
     apr_status_t rv;
     rng = apr_random_standard_new(p);
     do {
