commit 1f8e48dbabdaf65a0f8946bab5fa794e4580258d
Author: Daniel Muey <dan@cpanel.net>
Date:   Mon Mar 26 08:32:22 2018 -0500

    Apply 0003-silence-long-lost-pids.patch

diff --git a/server/mpm/event/event.c b/server/mpm/event/event.c
index 31fa04a..1179f9d 100644
--- a/server/mpm/event/event.c
+++ b/server/mpm/event/event.c
@@ -2944,7 +2944,7 @@ static void server_main_loop(int remaining_children_to_start, int num_buckets)
                 /* Great, we've probably just lost a slot in the
                  * scoreboard.  Somehow we don't know about this child.
                  */
-                ap_log_error(APLOG_MARK, APLOG_WARNING, 0,
+                ap_log_error(APLOG_MARK, APLOG_DEBUG, 0,
                              ap_server_conf, APLOGNO(00488)
                              "long lost child came home! (pid %ld)",
                              (long) pid.pid);
diff --git a/server/mpm/prefork/prefork.c b/server/mpm/prefork/prefork.c
index 3fb3284..752e3e3 100644
--- a/server/mpm/prefork/prefork.c
+++ b/server/mpm/prefork/prefork.c
@@ -991,7 +991,7 @@ static int prefork_run(apr_pool_t *_pconf, apr_pool_t *plog, server_rec *s)
                  * scoreboard.  Somehow we don't know about this
                  * child.
                  */
-                ap_log_error(APLOG_MARK, APLOG_WARNING,
+                ap_log_error(APLOG_MARK, APLOG_DEBUG,
                             0, ap_server_conf, APLOGNO(00167)
                             "long lost child came home! (pid %ld)", (long)pid.pid);
             }
diff --git a/server/mpm/worker/worker.c b/server/mpm/worker/worker.c
index 7804efc..afc401f 100644
--- a/server/mpm/worker/worker.c
+++ b/server/mpm/worker/worker.c
@@ -1619,7 +1619,7 @@ static void server_main_loop(int remaining_children_to_start, int num_buckets)
                 /* Great, we've probably just lost a slot in the
                  * scoreboard.  Somehow we don't know about this child.
                  */
-                ap_log_error(APLOG_MARK, APLOG_WARNING, 0,
+                ap_log_error(APLOG_MARK, APLOG_DEBUG, 0,
                              ap_server_conf, APLOGNO(00291)
                              "long lost child came home! (pid %ld)",
                              (long)pid.pid);
