From 7a234eb1182a620834a0c49965315f0597ac1f5d Mon Sep 17 00:00:00 2001
From: Cory McIntire <cory@cpanel.net>
Date: Wed, 1 Apr 2020 11:24:22 -0500
Subject: [PATCH 03/20] Add cPanel layout

---
 config.layout | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/config.layout b/config.layout
index a29351e..e5ce758 100644
--- a/config.layout
+++ b/config.layout
@@ -419,3 +419,26 @@
     proxycachedir:   ${localstatedir}/cache/apache2
 </Layout>
 
+# cPanel layout
+<Layout cPanel>
+    prefix:        /usr
+    exec_prefix:   ${prefix}
+    bindir:        ${prefix}/bin
+    sbindir:       ${prefix}/sbin
+    libdir:        ${prefix}/lib
+    libexecdir:    ${prefix}/libexec
+    mandir:        ${prefix}/man
+    sysconfdir:    /etc/apache2/conf
+    datadir:       ${prefix}/share/apache2
+    installbuilddir: ${libdir}/apache2/build
+    errordir:      ${datadir}/error
+    iconsdir:      ${datadir}/icons
+    htdocsdir:     /var/www/html
+    manualdir:     ${datadir}/manual
+    cgidir:        /var/www/cgi-bin
+    includedir:    ${prefix}/include/apache2
+    localstatedir: /var
+    runtimedir:    /var/run/apache2
+    logfiledir:    ${localstatedir}/log/apache2
+    proxycachedir: ${localstatedir}/cache/apache2/proxy
+</Layout>
-- 
2.42.0

