From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cory McIntire <cory@cpanel.net>
Date: Thu, 12 Jul 2018 15:51:37 -0500
Subject: [PATCH 15/21] Increase random seed size

---
 server/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/core.c b/server/core.c
index 22b6897..f9fe5e3 100644
--- a/server/core.c
+++ b/server/core.c
@@ -5546,7 +5546,7 @@ AP_CORE_DECLARE(void) ap_random_parent_after_fork(void)
 
 AP_CORE_DECLARE(void) ap_init_rng(apr_pool_t *p)
 {
-    unsigned char seed[8];
+    unsigned char seed[1024];
     apr_status_t rv;
     rng = apr_random_standard_new(p);
     do {
