From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Julian Brown <julian.brown@cpanel.net>
Date: Tue, 9 Jun 2020 13:07:33 -0500
Subject: [PATCH 20/21] Add instructions to install elinks.

---
 support/apachectl.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/support/apachectl.in b/support/apachectl.in
index 07c9912..21033a1 100644
--- a/support/apachectl.in
+++ b/support/apachectl.in
@@ -101,6 +101,9 @@ fi
 function checklynx() {
 if [ "$LYNX" = "none" ]; then
    echo "The 'links' package is required for this functionality."
+   echo "To use this feature, the typical way to add solve this would be to:"
+   echo "dnf config-manager --set-enabled powertools || dnf config-manager --set-enabled PowerTools"
+   echo "dnf -y install elinks"
    exit 8
 fi
 }
