From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Mullin <tim@cpanel.net>
Date: Wed, 18 Mar 2020 12:11:30 -0500
Subject: [PATCH 21/21] Update apxs to use the correct path for top_builddir

---
 support/apxs.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/support/apxs.in b/support/apxs.in
index d05d3f6..428d640 100644
--- a/support/apxs.in
+++ b/support/apxs.in
@@ -45,6 +45,7 @@ my $includedir     = get_vars("includedir");
 my $CFG_INCLUDEDIR = eval qq("$includedir");
 my $CFG_CC         = get_vars("CC");
 my $libexecdir     = get_vars("libexecdir");
+my $libdir         = get_vars("libdir");
 my $CFG_LIBEXECDIR = eval qq("$libexecdir");
 my $sbindir        = get_vars("sbindir");
 my $CFG_SBINDIR    = eval qq("$sbindir");
@@ -276,6 +277,7 @@ if ($opt_g) {
     $data =~ s|%TARGET%|$CFG_TARGET|sg;
     $data =~ s|%PREFIX%|$prefix|sg;
     $data =~ s|%INSTALLBUILDDIR%|$installbuilddir|sg;
+    $data =~ s|%LIBDIR%|$libdir|sg;
 
     my ($mkf, $mods, $src) = ($data =~ m|^(.+)-=#=-\n(.+)-=#=-\n(.+)|s);
 
@@ -588,7 +590,7 @@ __DATA__
 
 builddir=.
 top_srcdir=%PREFIX%
-top_builddir=%PREFIX%
+top_builddir=%LIBDIR%/apache2
 include %INSTALLBUILDDIR%/special.mk
 
 #   the used tools
