commit f9250018a1ed5ca357ecad14877565dc91ed2644
Author: Daniel Muey <dan@cpanel.net>
Date:   Mon Mar 26 08:32:17 2018 -0500

    Apply httpd-2.4.1-deplibs.patch

diff --git a/configure.in b/configure.in
index 761e836..06fe5f2 100644
--- a/configure.in
+++ b/configure.in
@@ -756,9 +756,9 @@ APACHE_HELP_STRING(--with-suexec-umask,umask for suexec'd process),[
 
 dnl APR should go after the other libs, so the right symbols can be picked up
 if test x${apu_found} != xobsolete; then
-  AP_LIBS="$AP_LIBS `$apu_config --avoid-ldap --link-libtool --libs`"
+  AP_LIBS="$AP_LIBS `$apu_config --avoid-ldap --link-libtool`"
 fi
-AP_LIBS="$AP_LIBS `$apr_config --link-libtool --libs`" 
+AP_LIBS="$AP_LIBS `$apr_config --link-libtool`" 
 APACHE_SUBST(AP_LIBS)
 APACHE_SUBST(AP_BUILD_SRCLIB_DIRS)
 APACHE_SUBST(AP_CLEAN_SRCLIB_DIRS)
